data/binding/queue.go

```diff
@@ -1,30 +1,33 @@
 package binding
 
 import (
-	"sync"
+	"runtime"
 
-	"fyne.io/fyne/v2/internal/async"
+	"fyne.io/fyne/v2"
 )
 
-var (
-	once  sync.Once
-	queue *async.UnboundedFuncChan
-)
+var mainGoroutineID uint64
 
-func queueItem(f func()) {
-	once.Do(func() {
-		queue = async.NewUnboundedFuncChan()
-		go func() {
-			for f := range queue.Out() {
-				f()
-			}
-		}()
-	})
-	queue.In() <- f
+func init() {
+	runtime.LockOSThread()
```

Comment:
With other runtime.LockOSThreads in the driver, can we be sure that this one will lock the same main thread? (ie is is a guarantee of Go that all init funcs call on the OS main thread, and redundant LockOSThread calls elsewhere will just be no-ops instead of getting a different Goroutine?)

Reply:
All init functions always run before the main function so there really are no other threads to run on. However, I have no idea how multiple calls to runtime.LockOSThread() behave.